File: Python/AdventDay1.py
def main():
    print("Python")
    with open('C:\\Users\\21310408\\Desktop\\Codes\\AdventOfCode2021\\inputDay1.txt') as f:    
        stringData = f.readlines()
    data = []
    for i, num in enumerate(stringData):
        data.append(int(num))
    numLarger = 0
    for i, num in enumerate(data):
        if i > 0:
            if num > data[i-1]:
                numLarger += 1
    print(f'Part 1:\nIncreases {numLarger} times.')
    numLarger = 0
    for i, num in enumerate(data):
        if i > 2:
            if num > data[i-3]:
                numLarger += 1
    print(f'Part 2:\nIncreases {numLarger} times.')
if __name__ == '__main__':
    main()

File: C/AdventDay1.c
#include <stdio.h> 
void main() {
	printf("C\n");
	FILE *file = fopen("C:\\Users\\21310408\\Desktop\\Codes\\AdventOfCode2021\\inputDay1.txt", "r");
	int data[2000];
	int dataSize = sizeof(data)/sizeof(data[0]);
	int numIncreased = 0;
	for (int i=0; i<dataSize; i++) {
		fscanf(file, "%d", & data[i]);
	}
	fclose(file);
	for (int i=1; i<dataSize; i++) {
		if (data[i]>data[i-1]) {
			numIncreased++;
		}
	}
	printf("Part 1:\nIncreases %d times.\n", numIncreased);
	numIncreased = 0;
	for (int i=3; i<dataSize; i++) {
		if (data[i]>data[i-3]) {
			numIncreased++;
		}
	}
	printf("Part 2:\nIncreases %d times.\n", numIncreased);
}